### PR TITLE
exception: print exception cause and location

### PR DESCRIPTION
**Pull request by abonislawski**

### CODE REVIEW

File: src/arch/xtensa/include/arch/init.h [hunk 1]
```diff
@@ -46,9 +46,14 @@
...
 {
+	uintptr_t epc1;
+
+	__asm__ __volatile__("rsr %0, EPC1" : "=a" (epc1) : : "memory");
+
 	/* now panic and rewind 8 stack frames. */
 	/* TODO: we could invoke a GDB stub here */
-	panic_rewind(SOF_IPC_PANIC_EXCEPTION, 8 * sizeof(uint32_t), NULL);
+	panic_rewind(SOF_IPC_PANIC_EXCEPTION, 8 * sizeof(uint32_t),
+		     NULL, &epc1);
```
Comment (mmaka1) [inline src/arch/xtensa/include/arch/init.h]: No need to do a redirection and pass ptr here (suggests that it is "inout", we should use more const btw). It is already a ptr, so passing 0 (=NULL) in panic (non-exception) case is OK.

File: tools/coredumper/sof-coredump-reader.py [hunk 2]
```diff
@@ -49,6 +49,51 @@ def stdout_print(*args, **kwargs):
 	]]
 ]
 
+# Exception casues:
+# [Exception cause, excvaddr loaded]
+EXCCAUSE_CODE = [
```
Comment (mmaka1) [inline tools/coredumper/sof-coredump-reader.py]: Not sure if dictionary (for potentially non-cont values) would be more appropriate than just an array indexed by the exception code.

File: tools/coredumper/sof-coredump-reader.py [hunk 3]
```diff
@@ -690,9 +734,17 @@ def __init__(self, args):
 			stdoutPrint("set *0x{:08x}=0x{:08x}\n"
 				.format(addr, dw))
 
+		if self.core_dump.exccause:
+			verbosePrint("# exccause: " + EXCCAUSE_CODE[self.core_dump.exccause][0])
```
Comment (mmaka1) [inline tools/coredumper/sof-coredump-reader.py]: A header like '\*EXCEPTION\*' would be a nice way to get an attention and highlight the next part.

File: tools/coredumper/sof-coredump-reader.py (hunk 2 above)
Comment (abonislawski) [reply tools/coredumper/sof-coredump-reader.py]: Sure, done

File: tools/coredumper/sof-coredump-reader.py (hunk 3 above)
Comment (abonislawski) [reply tools/coredumper/sof-coredump-reader.py]: Done

File: src/arch/xtensa/include/arch/init.h (hunk 1 above)
Comment (abonislawski) [reply src/arch/xtensa/include/arch/init.h]: Unfortunately it is the only working version. I have started with a standard passing by value but it result in a strange behaviour like changing the value when in panic_rewind().

File: src/include/sof/panic.h [hunk 4]
```diff
@@ -37,7 +37,7 @@ struct sof_ipc_panic_info;
 
 void dump_panicinfo(void *addr, struct sof_ipc_panic_info *panic_info);
 void panic_rewind(uint32_t p, uint32_t stack_rewind_frames,
-		  struct sof_ipc_panic_info *panic_info);
+		  struct sof_ipc_panic_info *panic_info, uintptr_t *epc1);
```
Comment (lgirdwood) [inline src/include/sof/panic.h]: epc1 needs to be renamed to data since this is a generic non xtensa header.

File: src/lib/panic.c [hunk 5]
```diff
@@ -52,7 +52,7 @@ void dump_panicinfo(void *addr, struct sof_ipc_panic_info *panic_info)
 
 /* panic and rewind stack */
 void panic_rewind(uint32_t p, uint32_t stack_rewind_frames,
-		  struct sof_ipc_panic_info *panic_info)
+		  struct sof_ipc_panic_info *panic_info, uintptr_t *epc1)
```
Comment (lgirdwood) [inline src/lib/panic.c]: ditto, rename to data in all generic code, but fine to use epc1 in xtensa code.

File: src/include/sof/panic.h (hunk 4 above)
Comment (abonislawski) [reply src/include/sof/panic.h]: Ok, done

File: src/lib/panic.c (hunk 5 above)
Comment (abonislawski) [reply src/lib/panic.c]: done

**Pull request merged**